### PR TITLE
fixes #12707 - index by content view version

### PR DESCRIPTION
**Pull request by thomasmckay**

### CODE REVIEW

**Comment by daviddavis** [comment]
Isn't this already being filtered here: http://git.io/vRHqN.

Also, I am wondering if hammer should be looking up the version at all when try to get the repositories for a content view. In the API at least, if you pass the content_view_id in, it lists the unpublished repos attached to the content view. See: http://git.io/vRHmZ.

**Comment by thomasmckay** [comment]
@daviddavis - Yes, however when running the hammer command the content_view_id is always going to be present and thus the query gets set by the method I modified.
```
hammer repository list --organization "Mega Corporation" --content-view "virt-who from sat-tools" --content-view-version 3.0
```

**Comment by daviddavis** [comment]
It should be the same query object though right? So then why do we need to call `query = query.where(:content_view_version_id => content_view_version_id)` twice?

**Comment by thomasmckay** [comment]
@daviddavis - Let me set up the test environment again and try. Perhaps it's an ordering thing on the query? I can only say right now that w/o my change it doesn't work. I think that last query w/ the join is stepping on things.

**Comment by daviddavis** [comment]
I suspect this change to repositories_controller should do the same thing without filtering by content_view_version_id twice:

https://gist.github.com/daviddavis/05d993a2ea3b62192d10

**Comment by thomasmckay** [comment]
@daviddavis - Yes, that works as well. I'll fix the rubocop and update this PR. Thanks!

File: app/controllers/katello/api/v2/repositories_controller.rb [hunk 1]
```diff
@@ -74,15 +73,28 @@ def index_relation
...
 
+      query = index_relation_content_view(query)
+      query = index_relation_environment(query)
+
+      query
+    end
+
+    def index_relation_product(query)
+      query = query.joins(:product).where("#{Product.table_name}.organization_id" => @organization) if @organization
+      query.where(:product_id => @product.id) if @product
```
Comment (daviddavis) [inline app/controllers/katello/api/v2/repositories_controller.rb]: This method returns nil if @product evaluates to false. Is that what you want?

**Comment by thomasmckay** [comment]
@daviddavis apologies, fixed now

**Comment by daviddavis** [comment]
APJ

**Comment by ehelms** [comment]
[test]

**Pull request merged**